GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -79,11 +79,23 @@ public async Task LaunchBuildServerInfoFetchOperationAsync()
 
             var fullDayObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler, DateTime.Today - TimeSpan.FromDays(3));
             var fullObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler);
-            var fromNowObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler, DateTime.Now);
 
             bool anyBuilds = false;
             var delayObservable = Observable.Defer(() => Observable.Empty<BuildInfo>()
                                                                    .DelaySubscription(anyBuilds ? ShortPollInterval : LongPollInterval));
+
+            var shouldLookForFinishedBuilds = false;
+            DateTime nowFrozen = DateTime.Now;
+            var fromNowObservable = Observable.Defer(() =>
+            {
+                // All finished builds have already been retrieved,
+                // so looking for new finished builds make sense only if running builds have been found previously
```

Comment:
Then we should rather append `fromNowObservable` to `runningBuildsObservable`, right?

If so, we could use single `anyBuilds` control variable, because `shouldLookForFinishedBuilds` would always have same value as `anyBuilds`

Reply:
>Then we should rather append fromNowObservable to runningBuildsObservable, right?

That's what I have imagined first but I decided against it because in this case it will run a lot more often (i.e. every 10s after running build check instead every minutes) and I will be obliged to add a lot more complicated logic (if possible) to keep this interval of 1 call/minutes which is enough for me...

Reply:
True, did not realize it at first glance.